packages/atclient/src/metadata.c

```diff
@@ -677,263 +681,377 @@ exit: {
 }
 }
 
-int atclient_atkey_metadata_to_protocolstr(const atclient_atkey_metadata metadata, char *metadatastr,
-                                           const size_t metadatastrlen, size_t *metadatastrolen) {
-  int ret = 1;
+size_t atclient_atkey_metadata_protocol_strlen(const atclient_atkey_metadata *metadata) {
```

Comment:
The other big metadata change, protocol_strlen was added to allow for faster computation of the notify string.